### PR TITLE
Ampersand debugger

### PR DESCRIPTION
**Pull request by andrewrota**

### CODE REVIEW

File: adaptors/ampersand/base_view.js [hunk 1]
```diff
@@ -122,6 +127,174 @@ var BaseView = AmpersandView.extend({
...
+    return data;
+  },
+
+  /**
+   * Sets the state to the given data
+   * @param {Object|Array} data Object to set state to
+   */
+  setState: function(data) {
+    var dataObj = this.serialize();
+    if (dataObj.reset) {
```
Comment (danieluhl) [inline adaptors/ampersand/base_view.js]: check reset and set for type 'function' for consistency

File: adaptors/ampersand/base_collection.js [hunk 2]
```diff
@@ -14,8 +16,110 @@ var eventBubbler = require('./event_bubbler');
...
+      if (typeof this[key] === 'function' && blacklist[key] !== true) {
+        result.push({
+          name: key,
+          fn: this[key],
+          inherited: (key in BaseCollection.prototype)
+        });
+        this[key] = getTrackableFunction(this, key, trackedFunctions);
+      }
+    }
+    return result;
```
Comment (MattDeeg) [inline adaptors/ampersand/base_collection.js]: Does this need the same Object.getOwnPropertyNames treatment that the Backbone version did?

Comment (andrewrota) [inline adaptors/ampersand/base_collection.js]: Yes, we should add that in.

**Comment by danieluhl** [comment]
Looks great!

**Pull request merged**